### DISCUSSION TITLE
The old Prefix/Postfix options should be brought back for the Stat Panel

### DISCUSSION THREAD

**Discussion by Ultraseamus** [grafana/grafana#39533] [Dashboards, Panels & Visualizations]
With Singlestat you used to be able to configure a prefix and postfix in addition to configuring the format of your numeric value. So you could do something like this:

"$5,999.99 Pretax"

In the new world, the only way to add custom prefixes/postfixes is with a custom unit formatter. But this means that you have to give up all control over how your numeric value is formatted. The example above becomes:

"5999.99 Pretax"

I could come up with other examples, but I think there is not much debate over pre/postfixes being useful in the Stat panel. And I think my example above shows that the current method of adding your own pre/postfix is basically non-functional. It was a clear regression.

If customizable pre/postfixes are a thing that Grafana wants to support, the current approach does not cut it.

I would either say to allow the chaining of formatting options (like let me format it as "Dollars" then format it with a postfix), which seems overly complex. Or just bring back the old controls.

As I understand it, those options were removed to try and keep the config free of clutter; but IMO, pre/postfixes are a basic enough feature when it comes to displaying numeric values, that they should definitely be given a few lines in Value configuration.

**Comment by pastansberry-clgx** [comment]
I previously used the "postfix" option to correct "10.4 week" (singular) to "10.4 weeks" (plural) in my uptime stat, because it drove my OCD crazy that more than one week was still listed as WEEK.

**Comment by Hantuch** [comment]
Would love to see this, why so apprehensive to customization???

**Comment by tjb36** [comment]
Agreed. I have an ultrahigh vacuum pressure reading in millibar, with values around 10^-11. So I need to use "Scientific Notation" as the unit, but then cannot add the "**mbar**" unit as well. I used to solve this with "postfix" in previous Grafana versions:
![Clipboard04](https://github.com/grafana/grafana/assets/43373700/53ae1cf9-9490-4823-89f1-1cc87992cae1)

Is there a way around it?

**Reply by ryantxu** [reply]
@tjb36 -- you can always use custom units to force the encoding you want:
https://grafana.com/docs/grafana/latest/panels-visualizations/configure-standard-options/#custom-units

The UX is pretty weird, you type into the picker "si:mbar" and will get si units with mbar suffix
<img width="254" alt="image" src="https://github.com/grafana/grafana/assets/705951/912daec0-2909-4744-913c-9bd0aed38dbf">

**Comment by Devineavenger** [comment]
Why was this removed? Bring it back please

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). This was added in https://github.com/grafana/grafana/issues/63528 Feel free to make an [issue](https://github.com/grafana/grafana/issues) if you cannot find a solution and would like to request the feature.

**Reply by tjb36** [reply]
You say "this was added", but I cannot see any option for it - can you confirm this, or provide clarity on where the option is located?
